vendor/github.com/virtblocks/virtblocks/go/rust/pkg/devices/BUILD.bazel

```diff
@@ -0,0 +1,12 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["memballoon.go"],
+    cgo = True,
+    clinkopts = ["-Lvendor/github.com/virtblocks/virtblocks/c/rust/.libs -lvirtblocks_c_rust -pthread -lm -ldl"],
+    copts = ["-Ivendor/github.com/virtblocks/virtblocks/c/rust"],
```

Comment:
How this is supposed to work? At least as far as I can see, you've only imported a single source file from the Virt Blocks project, `go/rust/pkg/devices/memballoon.go`, but that's only (a part of) the Go → Rust bindings: in order to build them successfully, you need the full Rust implementation as well as the C → Rust bindings.

Is this PR a work in progress, or am I simply missing something? O:-)

Reply:
It still WIP, I saw that I need first to compile the Rust code, after and after use this library under the C binding code, it like increase the complexity.

Reply:
Can Bazel just call out to the `Makefile` that's already present in the source directory? That already takes care of compiling the Rust and C bits before getting to the Go part.

Also is it necessary to provide the C compiler flags explicitly despite the fact that they're already specified in the Go source as `#cgo` comments? From the paths I assume it might be caused by the fact that you're vendoring the code...

Reply:
- bazel runs inside of the sandbox, it maybe can compile the code inside of it, but we will need to install `cargo` tool inside of our container and it will need additional plays to bring all dependencies into the sandbox on build time.
- You can provide C flags via bazel target

The best option will be(I do not consider to use go:)):
- bring rust code and compile it via https://github.com/bazelbuild/rules_rust/blob/master/docs/index.md#rust_library
- bring all headers via bazel target
- compile go code with rust library and C headers